### PR TITLE
unet time embedding activation function

### PR DESCRIPTION
**Pull request by williamberman**
See PR comments

### CODE REVIEW

File: src/diffusers/models/unet_2d_condition.py [hunk 1]
```diff
@@ -634,6 +653,9 @@ def forward(
             else:
                 emb = emb + class_emb
 
+        if self.time_embed_act is not None:
+            emb = self.time_embed_act(emb)
+
```
Comment (williamberman) [inline src/diffusers/models/unet_2d_condition.py:658]: optional activation of time embeddings once at at the beginning of the unet

File: src/diffusers/models/unet_2d_condition.py [hunk 2]
```diff
@@ -261,6 +266,20 @@ def __init__(
         else:
             self.class_embedding = None
 
+        if time_embedding_act_fn is not None:
+            if act_fn == "swish":
+                self.time_embed_act = lambda x: F.silu(x)
+            elif act_fn == "mish":
+                self.time_embed_act = nn.Mish()
+            elif act_fn == "silu":
+                self.time_embed_act = nn.SiLU()
```
Comment (sayakpaul) [inline src/diffusers/models/unet_2d_condition.py:275]: Can't we do?

```py
if act_fn in ["swish", "silu"]:
    self.time_embed_act = nn.SiLU()
```

File: src/diffusers/models/unet_2d_condition.py (hunk 1 above)
Comment (sayakpaul) [reply src/diffusers/models/unet_2d_condition.py:658]: Out of curiosity. 

Is it being used in the private fork?

**Review by sayakpaul — APPROVED** [review]

File: src/diffusers/models/unet_2d_condition.py (hunk 1 above)
Comment (williamberman) [reply src/diffusers/models/unet_2d_condition.py:658]: yessir!

File: src/diffusers/models/unet_2d_condition.py (hunk 2 above)
Comment (williamberman) [reply src/diffusers/models/unet_2d_condition.py]: Yes I would hope we could :) This is how it's done in a few other places in the code base so I'd like to leave it this way for now and do a follow up including a refactor of all the dispatches to the different activation functions

**Pull request merged**